src/core_modules/capture-core/components/Pages/New/RegistrationDataEntry/RegistrationDataEntry.epics.js

```diff
@@ -42,3 +45,42 @@ export const completeSavingNewTrackedEntityTypeEpic: Epic = (action$: InputObser
             return empty();
         }),
     );
+
+export const startSavingNewTrackedEntityInstanceWithEnrollmentEpic: Epic = (action$: InputObservable, store: ReduxStore) =>
+    action$.pipe(
+        ofType(registrationFormActionTypes.NEW_TRACKED_ENTITY_INSTANCE_WITH_ENROLLMENT_SAVE_START),
+        map(() => {
+            const { orgUnitId, programId } = store.value.currentSelections;
+            const enrollmentFormValues = store.value.dataEntriesFieldsValue['newPageDataEntryId-newEnrollment'] || {};
+            const scope = getScopeFromScopeId(programId);
```

Comment:
To me it makes more sense to use `getTrackerProgramThrowIfNotFound` here, since the `programId` is always a tracker program. This will also simplify the code a bit because you don't need to check `scope instanceof TrackerProgram` on line 68.

Reply:
Right thanks. That makes sense.